tsconfig.json

```
@@ -0,0 +1,14 @@
+{
+  "files": ["src/index.js"],
+  "compilerOptions": {
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "allowImportingTsExtensions": true,
+    "allowJs": true,
+    "checkJs": true,
+    "outDir": "dist/esm",
+    "target": "ES2022",
```

[Strict mode](https://www.typescriptlang.org/tsconfig/#strict) is a strong recommendation for TypeScript projects:

```suggestion
    "strict": true,
    "target": "ES2022",
```

If the codebase/team isn't ready for it just yet, there should be a TODO for adding it in.

---

I think this is an open question. It might be recommended, but I've found it to be overly cumbersome and I'm not sold that the value is worth the pain. We can follow up on that question later.